### DISCUSSION TITLE
Outcome transform in case of data with task indices

### DISCUSSION THREAD

**Discussion by sheikhahnaf** [meta-pytorch/botorch#2657] [Q&A]
Hi,
I am trying to use input and outcome transform in multitaskgp having task indices. In my case, I am using **botorch.models.transforms.outcome.Standardize**. But it requires **m arg** as dimension of output. But, with task indices, the ouput is flattened and it does not have columns corresponding to tasks. How to get around this? Is there solution already existing for this?

**Comment by saitcakmak** [comment]
Hi @sheikhahnaf. Outcome transforms do not differentiate between different tasks in MTGPs. If you'd like to use standardize each task independently, you'd have to do this manually. In Ax, we do this using [`StratifiedStandardizeY`](https://github.com/facebook/Ax/blob/2d375ab9b5a1f65c1acd4ea0d9c950b479519eef/ax/modelbridge/transforms/stratified_standardize_y.py#L35)

**Reply by sheikhahnaf** [reply]
thank you so much @saitcakmak
